S4TF_Tutorial_6.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "\n",
+        "In S4TF we can implement this using the different Python Libraries through Python Interoperability. We can simply pass different transformations we would want to our dataset as a form of arguments and it will take care of applying it to the dataset during our training process."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CnMOZA_BgREI",
+        "colab_type": "text"
+      },
+      "source": [
+        "We perform following Image Augmentation techniques randomly on an image:\n",
+        "\n",
+        "*  Rotating the image\n",
+        "* Transposing\n",
+        "* Cropping\n",
+        "* Flipping the image horizontally\n",
+        "* Adding border to image\n",
+        "* Adding Noise"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "QE-K4a2Ql8tx",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "func imageAugmentations(path: String) -> Tensor<Float> {\n",
+        "    \n",
+        "    var img = pil.Image.open(path)\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        img = img.rotate(45) // Rotate 45 degrees.\n",
+        "    }\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        img = img.transpose(pil.Image.TRANSPOSE) // Transpose.\n",
+        "    }\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        let h = Float(img.size[0])! / 4\n",
+        "        let w = Float(img.size[1])! / 4\n",
+        "        let cropBorder = Int(Python.min(h, w))\n",
+        "        img = pilImageOps.crop(img,  cropBorder) // Crop by half of shorter sides.\n",
+        "    }\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        img = img.transpose(pil.Image.FLIP_LEFT_RIGHT) // Flip.\n",
+        "    }\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        let h = Float(img.size[0])! / 10\n",
+        "        let w = Float(img.size[1])! / 10\n",
+        "        let borderSize = Int(Python.min(h, w))\n",
+        "        img = pilImageOps.expand(img, borderSize) // Add border.\n",
+        "    }\n",
+        "    \n",
+        "    var image = np.array(img, dtype: np.float32) * (1.0 / 255)\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        image = sk.util.random_noise(image) // Add noise.\n",
+        "    }\n",
+        "    \n",
+        "    image = np.array(image, dtype: np.float32)\n",
+        "    \n",
+        "    let imageTensor = Tensor<Float>(numpy: image)!\n",
```

Comment:
Return this directly instead of creating a new `let`.

Reply:
Returning Numpy Array directly was causing OOM or something like that. Further, it was misbehaving and returning out all black images. That's why I am returning image tensor which is easier to handle, as well as can be managed easily in the `resizedImage` function.